port_ocean/core/handlers/webhook/abstract_webhook_processor.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+from typing import Dict, Optional
+import asyncio
+from loguru import logger
+
+from .webhook_event import WebhookEvent, EventPayload
+
+
+class RetryableError(Exception):
+    """Base exception class for errors that should trigger a retry."""
+
+    pass
+
+
+class AbstractWebhookProcessor(ABC):
+    """Abstract base class for webhook processors."""
+
+    max_retries: int = 3
+    initial_retry_delay_seconds: float = 1.0
+    max_retry_delay_seconds: float = 30.0
+    exponential_base_seconds: float = 2.0
+
+    def __init__(self, event: WebhookEvent) -> None:
+        self.event = event
+        self.retry_count = 0
+
+    @property
+    def event_kind(self) -> Optional[str]:
+        """Extract event type from the webhook payload. Override if needed."""
+        return None
+
+    @property
+    def resource_id(self) -> Optional[str]:
+        """Extract resource identifier from the webhook payload. Override if needed."""
+        return None
```

Comment:
how will that be used? not sure I fully understand

Reply:
I thought we could use it for the updating port but it might be a placement for an AC

Reply:
Deleted